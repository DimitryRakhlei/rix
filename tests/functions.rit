int: sum0
  ->0

int: sum1 int a
  ->a

int: sum2 int a, int b
  ->a+b

int: sum3 int a, int b, int c
  print "sum3:"
  print "  arg 1: " + a
  print "  arg 2: " + b
  print "  arg 3: " + c
  ->a+b+c

: returnVoid
  print "returnVoid"



print "sum0: " + sum0
print "sum1: " + sum1 100
print "sum2: " + (sum2 100, 200)
print "  returned: " + (sum3 100, 200, 300)
//This only works with the parentheses in place.
print "func(func(), func()): " + (sum2 (sum2 10, 20), (sum2 300, 400))
echo "returnVoid: "
returnVoid

