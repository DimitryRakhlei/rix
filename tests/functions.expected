sum0: 0
sum1: 100
sum2: 300
sum3:
  arg 1: 100
  arg 2: 200
  arg 3: 300
  returned: 600
func(func(), func()): 730
returnVoid: returnVoid
